rfd/0131-adminitrative-actions-mfa.md

```diff
@@ -0,0 +1,488 @@
+
+1. User logs in with MFA.
+2. User updates their password or registered MFA devices.
+3. User connects to a service that requires Per-session MFA.
+
+As a result, the vast majority of Teleport actions do not require timely MFA
+verification. If an admin's login session is compromised, it can be used to
+perform attacks pertaining to privilege escalation, security downgrades, Auth
+connector changes, and other forms of cluster sabotage.
+
+Adding an MFA restriction to administrative actions limits this attack vector
+by re-verifying the user's identity promptly before performing any administrative
+action that could be abused.
+
+## Details
+
+### Server configuration
+
+Requiring MFA for admin actions will be made configurable through
+`cluster_auth_preference.admin_action_mfa: off | on`.
+
+When set to on, MFA will be required for admin actions for all users.
+
+Caveat: Built in roles will not require MFA to complete admin actions. This
+includes `tctl` operations performed directly on the auth server with the
+built in `Admin` role.
+
+If not set directly, this value will default to on or off based on the value of
+`clsuter_auth_preference.second_factor`:
+
+- `second_factor = off | optional` => `admin_action_mfa = off`
+- `second_factor = on | otp | webauthn` => `admin_action_mfa = on`
+
+#### Automated Use Cases
+
+In order to support automated use casessuch as the
+[Teleport Terraform Provider](https://goteleport.com/docs/management/guides/terraform-provider/)
+or [Access Request Plugins](https://goteleport.com/docs/access-controls/access-request-plugins/),
+we need to provide a way for specific user to bypass MFA for admin actions.
+
+We will do this with a matching role field `admin_action_mfa`. When set, this
+field will be prioritized over `cluster_auth_preference.admin_action_mfa`.
+We will not recommend setting this via roles outside of necessitating use cases.
+
+#### Backward Compatibility
+
+Once the Auth server begins to require MFA for admin actions, old clients with
+no way of providing MFA verification will fail to carry out admin actions.
+
+In order to maintain backwards compatibility between old clients and an upgraded
+server, we can't actually default `admin_action_mfa = on` as planned above.
+
+Instead, in the first major version of this feature (likely Teleport 14), it will
+always default to `off`. In the next major version (Teleport 15), the defaults
+above will be used.
+
+### Administrative Actions
+
+An administrative action will be defined as an action which creates, updates,
+or deletes a teleport resource. For example:
+
+- `tctl rm/create/edit`
+- `tctl users add/rm/reset`
+- Enrolling a new resource from the WebUI (Teleport Discover)
+- Modifying a role from the WebUI
+- Modifying a cloud upgrade window (at https://<tenant>.teleport.sh/web/support)
+
+To ensure this feature does not have unexpected effects, we will explicitly
+define which API requests are admin actions:
+
+gRPC:
```

Comment:
Maybe add trusted device operations to the list? https://github.com/gravitational/teleport/blob/45e1df013ec706e7b550059dad76caef97496be2/api/proto/teleport/devicetrust/v1/devicetrust_service.proto#L54